### ISSUE TITLE
BetterF3 Mod

### ISSUE THREAD

**Issue by FrapzHaash** [Fabulously-Optimized/fabulously-optimized#639]
### CurseForge link

https://www.curseforge.com/minecraft/mc-mods/betterf3

### CurseForge Mod Distribution

Allowed

### Modrinth link

https://modrinth.com/mod/betterf3

### Source/other link

https://github.com/cominixo/BetterF3

### Mod file size

1.707KB

### License

(any other license)

### What it does

BetterF3 is a mod that replaces Minecraft's original debug HUD with a highly customizable, more human-readable HUD. You can customize colors, position, add spacings, and more.

### Why should it be in the modpack

It is much cleaner and more beautiful than vanilla

### Why shouldn't it be in the modpack

idk really

### Additional details

It was rejected for about 1 year, but I think it should be considered again

**Comment by Madis0** [comment]
> It was rejected for about 1 year, but I think it should be considered again

Why? I still believe normal users should not need to use or customize the debug menu, as Sodium Extra has the coordinate and FPS labels separately available either way.

**Comment by FrapzHaash** [comment]
I think that there are things like the entities, the orientation and the light in the blocks are very important for many users

**Comment by Madis0** [comment]
Okay. I give you three theoretical users and you explain how the mod helps them:
1. 12-year old Bedwars player. They came to FO from PvP clients and are using mods that show PvP-related data, such as armor info.
2. 23-year old survival player. The player likes shaders and resource packs, also uses a minimap.
3. 45-year old creative player. Uses default resource pack, Litematica and WorldEdit to aid with building.
4. 60-year old creative player. Plays Minecraft with their children, who recommended FO for better performance, no additional mods.

**Comment by Leva25se** [comment]
> Okay. I give you four theoretical users and you explain how the mod helps them:
> 
>     1. 12-year old Bedwars player. They came to FO from PvP clients and are using mods that show PvP-related data, such as armor info.
For the pvp player F3 just takes up space on the screen (And sometimes it can interfere with the display of more useful information). All the information player needs is already displayed by other mods, which are much more customizable than F3.
>     2. 23-year old survival player. The player likes shaders and resource packs, also uses a minimap.
 1) The player uses bright shaders and looks at the sun/bright textures. It is impossible/very difficult for him to read the white font. He uses F3 to see fps, coordinates, but he sees a lot of information he doesn't need.
2) Since the player uses shaders and has to use F3 to see the lighting level of his house/build/mob farm, he may find it hard to find the information he needs in an overloaded F3 interface
>     3. 45-year old creative player. Uses default resource pack, Litematica and WorldEdit to aid with building.
1) The player, builds/designs a house, player built part of the roof of the stairs and wants to set/fill the rest of the roof, the same stairs, player need to use f3, information which has a lot of space on the screen, determine what refers to a block, determine what is the state of the block and what are its tags, and only then use the command to set blocks (Because F3 displays tags in addition to the state, it takes much longer to find the state than it would if the tags were not displayed).
>     4. 60-year old creative player. Plays Minecraft with their children, who recommended FO for better performance, no additional mods.
Some people who want to get from where they are now to a point with certain coordinates can get extremely confused by the coordinates and go the wrong way. Better F3 makes the coordinates colored, which makes it harder to mix up x and z (but still possible to go in the wrong direction).  It is much easier to use direction instead of coordinates. However, finding the meaning of directionality can be difficult for people who do not know English well.

**Comment by Julienraptor01** [comment]
For point 1 well OK so mod not needed

For point 1 of 2 well it's why sodium extra has show FPS and show coordinates
-> mod not needed
For point 2 of 2 there are shaders that can display light level iirc + disabling shaders for 2 secs takes one key press if configured
-> mod not needed, other ways are possible + some mods are dedicated to give this feature in a better way

For point 3 well, yes, when you learn you use it but as soon as the builder has done it quite some time, it start to remember 
-> vanilla F3 is enough for the learning time and can be toggled when needed

For point 4 it's purely opinionated
-> not needed and can be added cause opinion

My PoV is that yes, the debug screen can be useful for technical things that require debugging info but the F3 won't be used constantly anyway so if people want to enhance it fine, but it is never needed

**Comment by Leva25se** [comment]
> For point 1 well OK so mod not needed
> 
> For point 1 of 2 well it's why sodium extra has show FPS and show coordinates -> mod not needed For point 2 of 2 there are shaders that can display light level iirc + disabling shaders for 2 secs takes one key press if configured -> mod not needed, other ways are possible + some mods are dedicated to give this feature in a better way
> 
> For point 3 well, yes, when you learn you use it but as soon as the builder has done it quite some time, it start to remember -> vanilla F3 is enough for the learning time and can be toggled when needed
> 
> For point 4 it's purely opinionated -> not needed and can be added cause opinion
> 
> My PoV is that yes, the debug screen can be useful for technical things that require debugging info but the F3 won't be used constantly anyway so if people want to enhance it fine, but it is never needed One question to ask is why would we add a mod to enhance a debug feature which is only useful for specific task and shouldn't be used constantly if possible because it has a big performance hit?

About point 2.1: I did't see "Text Contrast", so now I agree with you.
For point 3: How can a player quickly determine the block face directions or directions in which the fence is connected? The easiest way is to use the block states from F3. But F3 displays tags, which are mostly only used by datapack creators and by some mod pack creators.
In addition to point 4 BetterF3 has a translation into other languages, which makes it much more convenient for people who don;t know English.
<details> 
<summary>Pefomance</summary>
Default F3
<img src="https://user-images.githubusercontent.com/125301618/234237231-232ea988-2ed4-43df-9f4a-6e220330598a.png">
<img src="https://user-images.githubusercontent.com/125301618/234237299-d46a1292-5be9-44b7-8e04-76da0203d82a.png">
Default BetterF3
<img src="https://user-images.githubusercontent.com/125301618/234237414-dc037635-7aad-4ac1-a475-47cebed8d8c5.png">
<img src="https://user-images.githubusercontent.com/125301618/234237437-1996235a-5416-4b91-92ea-f6a4f58e49b3.png">
BetterF3 with partially disabled information
<img src="https://user-images.githubusercontent.com/125301618/234237996-a47366f8-0992-468b-aeb7-7541c1d2eecd.png">
<img src="https://user-images.githubusercontent.com/125301618/234237979-814234bc-15be-43b2-a8e4-c051f18f588c.png">
Clean F3
<img src="https://user-images.githubusercontent.com/125301618/234239433-9c6b9de0-0b1b-45c7-a29c-b6fe11608252.png">
<img src="https://user-images.githubusercontent.com/125301618/234239470-308e92ab-f01e-424b-bff3-dc8dffe3c5c8.png">
</details>

**Comment by Madis0** [comment]
I agree with Julien:

1 - there are better mods for that (Armor HUD, etc)
2 - there are better mods for that (Light Level Display, etc)
3 - there are better mods for that (Jade, WTHIT, etc...)
4 - languages and colors are good, but again - I don't even want to _endorse_ the use of debug screen. Minecraft is such a rare game where people actually use one, true, but that doesn't mean it is the best way for those purposes anyway.

Rejected.

**Comment by Kichura** [comment]
tl;dr: This mod is pointless to have in practice and because it is a duplicate issue of the original issue, rejected.
